system/HTTP/SiteURI.php

```diff
@@ -0,0 +1,312 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\HTTP;
+
+use BadMethodCallException;
+use CodeIgniter\HTTP\Exceptions\HTTPException;
+use Config\App;
+
+/**
+ * URI for the application site
+ */
+class SiteURI extends URI
+{
+    /**
+     * The baseURL.
+     */
+    private string $baseURL;
+
+    /**
+     * The Index File.
+     */
+    private string $indexPage;
+
+    /**
+     * List of URI segments in baseURL and indexPage.
+     *
+     * If the URI is "http://localhost:8888/ci431/public/index.php/test?a=b",
+     * and the baseUR is "http://localhost:8888/ci431/public/", then:
+     *   $baseSegments = [
+     *       0 => 'ci431',
+     *       1 => 'public',
+     *       2 => 'index.php',
+     *   ];
+     */
+    private array $baseSegments;
+
+    /**
+     * List of URI segments after indexPage.
+     *
+     * The word "URI Segments" originally means only the URI path part relative
+     * to the baseURL.
+     *
+     * If the URI is "http://localhost:8888/ci431/public/index.php/test?a=b",
+     * and the baseUR is "http://localhost:8888/ci431/public/", then:
+     *   $segments = [
+     *       0 => 'test',
+     *   ];
+     *
+     * @var array
+     *
+     * @deprecated This property will be private.
+     */
+    protected $segments;
+
+    /**
+     * URI path relative to baseURL.
+     *
+     * If the baseURL contains sub folders, this value will be different from
+     * the current URI path.
+     */
+    private string $routePath;
+
+    public function __construct(App $configApp)
+    {
+        // It's possible the user forgot a trailing slash on their
+        // baseURL, so let's help them out.
+        $baseURL = rtrim($configApp->baseURL, '/ ') . '/';
+
+        $this->baseURL   = $baseURL;
+        $this->indexPage = $configApp->indexPage;
+
+        $this->setBaseSeegments();
+
+        // Check for an index page
+        $indexPage = '';
+        if ($configApp->indexPage !== '') {
+            $indexPage = $configApp->indexPage . '/';
+        }
+
+        $tempUri = $this->baseURL . $indexPage;
+        $uri     = new URI($tempUri);
+
+        if ($configApp->forceGlobalSecureRequests) {
+            $uri->setScheme('https');
+        }
+
+        $parts = parse_url((string) $uri);
+        if ($parts === false) {
+            throw HTTPException::forUnableToParseURI($uri);
+        }
+        $this->applyParts($parts);
+
+        $this->setPath('/');
+    }
+
+    /**
+     * Sets baseSegments.
+     */
+    private function setBaseSeegments(): void
```

Comment:
Seegments -> Segments

```suggestion
    private function setBaseSegments(): void
```

Reply:
Thanks! Fixed.